src/uct/tcp/sockcm/sockcm_md.c

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2017.  ALL RIGHTS RESERVED.
+ * Copyright (C) NVIDIA Corporation. 2019.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#include "sockcm_md.h"
+
+#define UCT_SOCKCM_MD_PREFIX              "sockcm"
+
+static ucs_config_field_t uct_sockcm_md_config_table[] = {
+  {"", "", NULL,
+   ucs_offsetof(uct_sockcm_md_config_t, super), UCS_CONFIG_TYPE_TABLE(uct_md_config_table)},
+  {NULL}
+};
+
+static void uct_sockcm_md_close(uct_md_h md);
+
+static uct_md_ops_t uct_sockcm_md_ops = {
+    .close                  = uct_sockcm_md_close,
+    .query                  = uct_sockcm_md_query,
+    .is_sockaddr_accessible = uct_sockcm_is_sockaddr_accessible,
+    .is_mem_type_owned      = (void *)ucs_empty_function_return_zero,
+};
+
+static void uct_sockcm_md_close(uct_md_h md)
+{
+    uct_sockcm_md_t *sockcm_md = ucs_derived_of(md, uct_sockcm_md_t);
+    ucs_free(sockcm_md);
+}
+
+ucs_status_t uct_sockcm_md_query(uct_md_h md, uct_md_attr_t *md_attr)
+{
+    md_attr->cap.flags         = UCT_MD_FLAG_SOCKADDR;
+    md_attr->cap.reg_mem_types = 0;
+    md_attr->cap.mem_type      = UCT_MD_MEM_TYPE_HOST;
+    md_attr->cap.max_alloc     = 0;
+    md_attr->cap.max_reg       = 0;
```

Comment:
maybe just `memset()` for the `md_attr`?

Reply:
not all fields are 0..